docs/USAGE.md

```diff
@@ -0,0 +1,282 @@
+complete implementation of the OpenTracing API.
+
+The auto-instrumentation provides a base you can build on by adding your own
+custom instrumentation. By using both instrumentation approaches, you'll be
+able to present a more detailed representation of the logic and functionality
+of your application, clients, and framework.
+
+1. Add the OpenTracing dependency to your project:
+    ```xml
+    <PackageReference Include="OpenTracing" Version="0.12.0" />
+    ```
+2. Obtain the `OpenTracing.Util.GlobalTracer` instance and create spans that
+automatically become child spans of any existing spans in the same context:
+    ```csharp
+    using OpenTracing;
+    using OpenTracing.Util;
+
+    namespace MyProject
+    {
+        public class MyClass
+        {
+            public static async void MyMethod()
+            {
+                // Obtain the automatically registered OpenTracing.Util.GlobalTracer instance
+                var tracer = GlobalTracer.Instance;
+
+                // Create an active span that will be automatically parented by any existing span in this context
+                using (IScope scope = tracer.BuildSpan("MyTracedFunctionality").StartActive(finishSpanOnDispose: true))
+                {
+                    var span = scope.Span;
+                    span.SetTag("MyImportantTag", "MyImportantValue");
+                    span.Log("My Important Log Statement");
+
+                    var ret = await MyAppFunctionality();
+
+                    span.SetTag("FunctionalityReturned", ret.ToString());
+                }
+            }
+        }
+    }
+    ```
+
+## Troubleshooting
+
+Check if you are not hitting one of the issues listed below.
+
+### IIS applications not instrumenting expected services
+
+Set the environment variable `OTEL_TRACE_DOMAIN_NEUTRAL_INSTRUMENTATION` to `true` - without it
+the CLR profiler can't instrument many libraries/frameworks under IIS.
+
+### Linux instrumentation not working
+
+The proper binary needs to be selected when deploying to Linux, eg.: the default Microsoft .NET images are
+based on Debian and should use the `deb` package, see the [Linux](#Linux) setup section.
+
+If you are not sure what is the Linux distribution being used try the following commands:
+```terminal
+$ lsb_release -a
+$ cat /etc/*release
+$ cat /etc/issue*
+$ cat /proc/version
+```
+
+### High CPU usage
+
+The default installation of auto-instrumentation enables tracing all .NET processes on the box.
+In the typical scenarios (dedicated VMs or containers), this is not a problem.
+Use the environment variables `OTEL_PROFILER_EXCLUDE_PROCESSES` and `OTEL_PROFILER_PROCESSES`
+to include/exclude applications from the tracing auto-instrumentation.
+These are ";" delimited lists that control the inclusion/exclusion of processes.
+
+### Custom instrumentation not being captured
+
+If the code accessing `GlobalTracer.Instance` executes before any auto-instrumentation is injected
```

Comment:
We need to double-check if this applies here too.

Reply:
It "applies" in the sense that we still give the possibility to the instrument using OpenTracing.

However, based on https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/129#discussion_r616875205 I think this section is not needed

@zacharycmontoya could you confirm, please?

Reply:
> It "applies" in the sense that we still give the possibility to the instrument using OpenTracing.

yeah, I meant in the sense that I'm not sure if this is still an issue or not.

Reply:
I'll remove that